### ISSUE TITLE
Missing iterator_pair.h for building mymodule in nest 2.20.0

### ISSUE THREAD

**Issue by niltonlk** [nest/nest-simulator#1478]
**Describe the bug**
When trying to make external-module after nest have being installed, an error message stating that iterator_pair.h was not found is displayed. In addition, copying the file found in libnestutil folder results in another error stating that ../nestkernel/source.h was not found.

**To Reproduce**
1. Install NEST release 2.20.0 without external module such as:
```
$ mkdir nest-make-dir
$ cd nest-make-dir
$ cmake -DCMAKE_INSTALL_PREFIX:PATH=<NEST_INSTALL_DIR> -Dwith-mpi=ON -Dwith-python=3 -DPYTHON_EXECUTABLE=`which python3.6` -DNOSETESTS=`which nosetests` -DCYTHON_EXECUTABLE=`which cython` -Dwith-boost=ON -Dwith-includes=<BOOST_INCLUDE_PATH> ../nest-simulator-2.20.0
$ source <NEST_INSTALL_DIR>/bin/nest_vars.sh
```
2. Try to install mymodule with:
```
$ cd ..
$ mkdir mymodule-make-dir
$ cd mymodule-make-dir
$ cmake ./nest-simualtor-2.20.0/example/MyModule
$ make
```
3. Output example-1:
```
from nest-simulator-2.20.0/examples/MyModule/mymodule.cpp:29:
<NEST_INSTALL_DIR>/include/nest/sort.h:37:10: fatal error: iterator_pair.h: No such file or directory
#include "iterator_pair.h"
```
4. Copy libnestutil/iterator_pair.h into <NEST_INSTALL_DIR>/include/nest and try to make mymodule again
```
$ cp -a nest-simulator-2.20.0/libnestutil/iterator_pair.h <NEST_INSTALL_DIR>/include/nest
$ make
```
5. Output example-2:
```
<NEST_INSTALL_DIR>/include/nest/iterator_pair.h:29:10: fatal error: ../nestkernel/source.h: No such file or directory
#include "../nestkernel/source.h"
```

**Expected behavior**
Expected that make command ends with no error

**Desktop/Environment (please complete the following information):**
 - OS: CentOS 7.7.1908
 - Shell: bash
 - Python-Version: Python 3.6
 - NEST-Version: nest-2.20
 - Installation: from source, see above (with MPI, with boost)
    Configuration summary not available as it has passed several weeks after NEST installation.

**Additional context**
Changing line 29 in the copied iterator_pair.h to:
```
#include "source.h"
```
Fix the issue!
Should this be included into 2.20.1 (#1455)?

The files mentioned in this issue seems to be fixed in the github master, however, trying to make mymodule after NEST have being installed leads to another error: SFML/Audio.hpp: No such file or directory.

**Comment by hakonsbm** [comment]
I have created a PR fixing this issue.

> The files mentioned in this issue seems to be fixed in the github master, however, trying to make mymodule after NEST have being installed leads to another error: SFML/Audio.hpp: No such file or directory.

To use the soundclick recording backend, SFML must be installed. CMake is supposed to check if it is installed, and not compile the pertaining files if it is not. But this apparently does not work properly. I will make an issue.

**Comment by clinssen** [comment]
The latter already has a PR #1512

**Comment by hakonsbm** [comment]
@clinssen Ah, that's good.

**Comment by heplesser** [comment]
Fixed by #1514.
